test/test_document.rb

```diff
@@ -417,6 +417,34 @@ def test_utf_16
           assert_equal(expected_xml, actual_xml)
         end
       end
+
+      class EachRecursiveTest < Test::Unit::TestCase
+        def test_each_recursive
+          xml_source = +'<?xml version="1.0" encoding="UTF-8" standalone="yes"?>'
+          xml_source << '<root foo="bar">'
+
+          100.times do
+            x_node_source = ''
+            100.times do
```

Comment:
Do we need `100` for this test?
Can we reduce it?

Can we use different number for outer `times` and inner `times` for easy to distinct a child problem and a descendant problem.